graylog2-server/src/main/java/org/graylog2/shared/buffers/processors/ProcessBufferProcessor.java

```diff
@@ -56,6 +57,7 @@
     private final ULID ulid;
     private final DecodingProcessor decodingProcessor;
     private final Provider<Stream> defaultStreamProvider;
+    private MessageEvent currentEvent;
```

Comment:
I think we should make this variable `volatile` to ensure cross-thread visibility.

Reply:
Good catch :+1: